src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -163,8 +163,8 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 	private _title: string = '';
 	private _titleSource: TitleEventSource = TitleEventSource.Process;
 	private _container: HTMLElement | undefined;
-	get container(): HTMLElement | undefined { return this._container; }
 	private _wrapperElement: (HTMLElement & { xterm?: XTermTerminal });
+	get domElement(): HTMLElement { return this._wrapperElement; }
```

Comment:
Could this be removed in favor of `xterm.raw.element` in `TerminalFIndContribution.xtermReady`?

Reply:
How about `xterm.raw.element.parentElement`? If not, anything beyond `parentElement` is too hacky

Reply:
Parent element is this [plain div here](https://github.com/microsoft/vscode/blob/70fa2da3646d385e6e07c834abe3f5b41eada4f7/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L899-L900C8) that can moved to be created in constructor too and exposed through the interface, also maybe is possible to remove it so the parent element is `_wrapperElement` itself
Or do you mean you prefer not to expose the dom element in the `terminalInstance` interface

Reply:
Any interfaces that make sense to remove from `ITerminalInstance` is a win, but we can leave it as it's not specifically fixing this issue.